packages/contracts-bedrock/test/scripts/VerifyOPCM.t.sol

```diff
@@ -223,4 +227,40 @@ contract VerifyOPCM_Run_Test is VerifyOPCM_TestInit {
         vm.expectRevert(VerifyOPCM.VerifyOPCM_Failed.selector);
         harness.run(address(opcm), true);
     }
+
+    /// @notice Tests that the script verifies all component contracts have the same contractsContainer address.
+    function test_verifyContractsContainerConsistency_succeeds() public {
+        // Coverage changes bytecode and causes failures, skip.
+        skipIfCoverage();
+
+        // Get the property references (which include the component addresses)
+        VerifyOPCM.OpcmContractRef[] memory propRefs = harness.getOpcmPropertyRefs(opcm);
+
+        // This should succeed with the current setup where all contracts have the same containerAddress.
+        harness.verifyContractsContainerConsistency(propRefs);
+    }
+
+    /// @notice Tests that the script reverts when contracts have different contractsContainer addresses.
+    function test_verifyContractsContainerConsistency_mismatch_reverts() public {
+        // Coverage changes bytecode and causes failures, skip.
+        skipIfCoverage();
+
+        // Create a different address to simulate a mismatch.
+        address differentContainer = address(0x9999999999999999999999999999999999999999);
+
+        // Get the opcmUpgrader address and modify its contractsContainer() return value
+        address upgrader = address(opcm.opcmUpgrader());
+
+        // Mock the contractsContainer() call to return a different address
+        vm.mockCall(
+            upgrader, abi.encodeCall(IOPContractsManagerUpgrader.contractsContainer, ()), abi.encode(differentContainer)
+        );
+
+        // Get the property references (which include the component addresses)
+        VerifyOPCM.OpcmContractRef[] memory propRefs = harness.getOpcmPropertyRefs(opcm);
+
+        // Now the consistency check should fail.
+        vm.expectRevert(VerifyOPCM.VerifyOPCM_ContractsContainerMismatch.selector);
+        harness.verifyContractsContainerConsistency(propRefs);
```

Comment:
We should make sure that this test is future proof incase some new `opcmDoesNewThing()` component is added.

I think a good approach would be to:
1. first get the `propRefs` array 
2. extract the `vm.mockCall()` and `vm.expectRevert()` into a utility function.
3. iterate over the `propRefs` array, so that we test to make sure that there's a failure no matter which one has an incorrect container.